plugin.js

```diff
@@ -40,8 +41,24 @@ const defaultProcessFn = (builtPolicy, htmlPluginData, $) => {
   // build the policy into the context attr of the csp meta tag
   metaTag.attr('content', builtPolicy);
 
-  // eslint-disable-next-line no-param-reassign
-  htmlPluginData.html = $.html();
+  const xmlMode = get(htmlPluginData, 'plugin.options.xhtml', false);
+  const rawTag = metaTag.get(0);
+  if (
+    xmlMode &&
+    (!haveCSP ||
+      (rawTag.startIndex !== undefined && rawTag.endIndex !== undefined))
+  ) {
+    const tag = $.html(metaTag, { xml: true });
+    const html = !haveCSP
+      ? htmlPluginData.html
+      : htmlPluginData.html.slice(0, rawTag.startIndex) +
+        htmlPluginData.html.slice(rawTag.endIndex + 1);
+    // eslint-disable-next-line no-param-reassign
+    htmlPluginData.html = html.replace(/(<head\b[^>]*>)/i, `$1${tag}`);
```

Comment:
Cut off the existing CSP meta tag if it exists, and add our CSP right after `<head>` (same as `metaTag.prependTo($('head'))` would do).